src/sagemaker/model.py

```diff
@@ -698,6 +703,9 @@ def deploy(
         kms_key=None,
         wait=True,
         data_capture_config=None,
+        async_inference_config=None,
+        serverless_inference_config=None,
+        inference_type="real_time",
```

Comment:
Is it necessary to have an inference_type? If Async + Serverless are eventually supported together, this might cause some issues, right? Can we instead just use the presence of the various fields to determine the 'type' when we need to?

If we want to be able to use defaults then maybe something like `is_async` is better? But I am also fine with making them required nested fields within the respective config objects themselves.

Reply:
I'm not quite sure about what the issues will be and could you give more details for the issues you're concerned? So for now, the change will support both async and serverless inference. The reason to have this `inference_type` is just like you said, it'll be used to generate default async/serverless endpoint config. 

Correct me if I'm wrong, I think the Sagemaker SDK will be easier to use compare to BOTO3 so I'm considering the scenario: if customers don't want to create and pass config objects and just want to use async/serverless inference with one line code, they can specify the `inference_type`. And we use default configuration to deploy an async/serverless endpoint. In this case, we can't tell what type of endpoints customers want to use without this arg. On the other side, if 'async_inference_config'/'serverless_inference_config' is not None, I can add some check here to determine the type of endpoint and ignore this arg.